core/dbt/include/global_project/macros/materializations/common/merge.sql

```diff
@@ -77,3 +82,23 @@
 {% macro default__get_delete_insert_merge_sql(target, source, unique_key, dest_columns) -%}
     {{ common_get_delete_insert_merge_sql(target, source, unique_key, dest_columns) }}
 {% endmacro %}
+
+
+{% macro default__get_insert_overwrite_merge_sql(target, source, dest_columns, predicates) -%}
+    {%- set predicates = [] if predicates is none else [] + predicates -%}
+    {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
+
+    merge into {{ target }} as DBT_INTERNAL_DEST
```

Comment:
i could live 1,000 more years and i would still not understand this DML...

Reply:
I have lived 2 days and I have read [the documentation](https://cloud.google.com/bigquery/docs/reference/standard-sql/dml-syntax#merge_statement), and I now understand this DML.

The key thing I was missing: `when not matched by source` is _always_ true here because we're using a constant-false predicate. I was errantly thinking that we were still merging on a `unique_key`, which we are not.

From the docs:
> If the merge_condition is FALSE, the query optimizer avoids using a JOIN. This optimization is referred to as a constant false predicate. A constant false predicate is useful when you perform an atomic DELETE on the target plus an INSERT from a source (DELETE with INSERT is also known as a REPLACE operation).

Cool!